packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -3747,7 +3795,57 @@ export function attach(
   }
 
   // React will switch between these implementations depending on whether
-  // we have any manually suspended Fibers or not.
+  // we have any manually suspended/errored-out Fibers or not.
+  function shouldErrorFiberAlwaysNull() {
+    return null;
+  }
+
+  const forceErrorForFiberIDs = new Map();
+  function shouldErrorFiberAccordingToMap(fiber) {
+    if (typeof setErrorHandler !== 'function') {
+      throw new Error(
+        'Expected overrideError() to not get called for earlier React versions.',
+      );
+    }
+
+    const id = getOrGenerateFiberID(fiber);
```

Comment:
I _think_ this should use `getFiberIDUnsafe` and if the ID returned is `null` then we should return `false`.

Reply:
I'm having trouble with this actually. For some reason, in certain scenario, the id would be null 



https://user-images.githubusercontent.com/783410/120501237-09df5580-c3f4-11eb-836e-074cc286d03f.mov

Reply:
yep you can change it to `getFiberIDUnsafe` and log in `ReactFiberBeginWork`

![image](https://user-images.githubusercontent.com/783410/120504392-ad316a00-c3f6-11eb-9b8c-463628ec40ec.png)

Reply:
Seems like it works okay for me when I test? To be clear, this is what I'm suggesting:

```js
  function shouldErrorFiberAccordingToMap(fiber) {
    if (typeof setErrorHandler !== 'function') {
      throw new Error(
        'Expected overrideError() to not get called for earlier React versions.',
      );
    }

    const id = getFiberIDUnsafe(fiber);
    if (id === null) {
      // If this Fiber is not mounted, we can ignore this request.
      // This statement should probably never actually be hit.
      return null;
    }

    let status = null;
    if (forceErrorForFiberIDs.has(id)) {
      status = forceErrorForFiberIDs.get(id);
      if (status === false) {
        clearErrorsForFiberID(id);
      }

      if (forceErrorForFiberIDs.size === 0) {
        // Last override is gone. Switch React back to fast path.
        setErrorHandler(shouldErrorFiberAlwaysNull);
      }
    }

    return status;
  }
```

Reply:
ahhhhhh, I `return false` instead!! that's why it keeps trying. Returning `null` works!

However, can you try to log when the id is null? I can see that the id would be null if I go through the same sequence as in the video I attached in my previous comment.

Reply:
Does the id being null make any practical difference?

I believe I added such a log and _didn't_ see anything being printed FWIW.